azure/services/scalesets/scalesets.go

```diff
@@ -655,9 +655,12 @@ func base64EncodedHash(vmss compute.VirtualMachineScaleSet) (string, error) {
 	// Setting Admin Password is not supported but an initial password is required for Windows
 	// Don't include it in the hash since it is generated and won't be the same each the spec is created (#1182)
 	tmpPass := vmss.VirtualMachineProfile.OsProfile.AdminPassword
+	tmpCustomData := vmss.VirtualMachineProfile.OsProfile.CustomData
 	vmss.VirtualMachineProfile.OsProfile.AdminPassword = nil
+	vmss.VirtualMachineProfile.OsProfile.CustomData = nil
 	defer func() {
 		vmss.VirtualMachineProfile.OsProfile.AdminPassword = tmpPass
+		vmss.VirtualMachineProfile.OsProfile.CustomData = tmpCustomData
```

Comment:
how do this affect scale up? if we're not giving the new bootstrap token to the model how do new nodes join after the initial token has expired?

Reply:
- If the replica count increases on MachinePool, then the AMP will PATCH with the updated token and capacity b/c it was caused by an action other than the token changing.
- If the model (image data, identities, etc) changes on MachinePool / AMP, then the VMSS will be PATCH'd with the updated model and token.

Does this make sense? Am I missing a scenario? I was concerned about this possibility too.

To be clear, I don't love this approach.

Reply:
Ah I see, this func only calculates the diff to determine if a PATCH is required, but it doesn't actually exclude these things from the actual content of the PATCH that is sent if there is a diff. Makes sense to me 👍 

I don't love the approach either but I think it's acceptable if it's very well commented so we don't look at this code later and wonder exactly the above.